packages/api/endpoints/rules.js

```diff
@@ -60,26 +64,29 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
+  const {
+    model = new models.Rule(),
+    dbClient = await getKnexClient(),
+  } = req.testContext || {};
+
+  let record;
+  const rule = req.body || {};
+  const name = rule.name;
+  rule.createdAt = Date.now();
+  rule.updatedAt = Date.now();
+
+  if (await model.exists(name)) {
+    return res.boom.conflict(`A record already exists for ${name}`);
+  }
+
   try {
-    const data = req.body;
-    const name = data.name;
-
-    const model = new models.Rule();
-
-    try {
-      await model.get({ name });
-      return res.boom.conflict(`A record already exists for ${name}`);
-    } catch (error) {
-      if (error instanceof RecordDoesNotExist) {
-        const record = await model.create(data);
-
-        if (inTestMode()) {
-          await addToLocalES(record, indexRule);
-        }
-        return res.send({ message: 'Record saved', record });
-      }
-      throw error;
-    }
+    await dbClient.transaction(async (trx) => {
+      const postgresRecord = await translateApiRuleToPostgresRule(rule, dbClient);
```

Comment:
It's subtle: but as this function isn't directly related to writing records, I would move it out above the transaction. The transaction should still fail if this function returns a `postgresRecord` that isn't compatible with Postgres anyway. 

I would make this change throughout the endpoint code

Reply:
Oh that makes sense to me. I will remove it outside of the transaction.

Reply:
[`3a15639` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/3a1563919a247377bb1f57c4fa1ff3599e49762d)